### PR TITLE
Reorganize file structure

### PR DESCRIPTION
**Pull request by KinanBab**
1. Put all protobuf definitions in one file
2. rename all BUILD and WORKSPACE file to .bazel
3. put the p4-sample programs in a separate directory under / since they will be shared by many components
4. put bmv2 json parsing cc file, protobufs, and test files, including expected files and python script, under src/bmv2
5. Make appropriate changes to package naming, cc includes and bazel loads, and test scripts
6. Re-write and improve veth_setup.sh and update p4-samples Makefiles.

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Make sure programs under p4-samples are visible to other
```
Comment (smolkaj) [inline BUILD.bazel]: Move to the p4-samples folder.

File: src/bmv2/BUILD.bazel [hunk 2]
```diff
@@ -12,37 +12,61 @@
...
+    name = "test",
+    srcs = ["test.cc"],
+    deps = [":bmv2_cc_proto"]
+)
 
+# Test rules: one per program in //p4-samples
 bmv2_protobuf_parsing_test(
     name = "port_table",
-    p4_program = "samples/port-table/table.p4",
+    p4_program = "//:p4-samples/port-table/table.p4",
```
Comment (smolkaj) [inline src/bmv2/BUILD.bazel]: After moving the BUILD file as suggested above, this will become `//p4-samples:port-table/table.p4` and similarly below.

**Review by smolkaj — APPROVED** [review]
LGTM. Just a small comment.

**Comment by KinanBab** [comment]
@smolkaj done. I will wait for stefan to approve as well before merging.

File: src/bmv2/bmv2.proto [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file contains protobuf definitions for parsing bmv2 json
+// files (outputs of p4c).
+// Some attributes here are intentionally un-constrained (e.g.
+// struct). This allows parsing of radically different possibilities
+// or types of constructs that may appear interchangably in that
+// attribute.
+// Further in our pipeline, we have some c++ code that traverses
```
Comment (stefanheule) [inline src/bmv2/bmv2.proto]: I would remove this last paragraph. It is just going to become stale/out-of-date, and it doesn't add much. We should explain the different phases in the place where we actually do it, not here.

File: src/bmv2/bmv2.proto [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+package p4_symbolic.bmv2;
+
+import "google/protobuf/struct.proto";
+
+// Specifies the overal structure of a p4 program json
+// file outputed by p4c with a bmv2 target.
+// Several fields may not be present in this specification
+// because they are irrelevant to our tool.
+// Furthermore, certain parts of the json program
```
Comment (stefanheule) [inline src/bmv2/bmv2.proto]: Same here, I would remove lines 35-45.

**Comment by KinanBab** [comment]
Renamed src to p4_symbolic. Path and package paths now are identical. 

Closes #9.

File: p4_symbolic/bmv2/test.cc [hunk 5]
```diff
@@ -26,10 +26,10 @@
 #include "google/protobuf/util/json_util.h"
 #include "google/protobuf/text_format.h"
 
-#include "src/protobuf/bmv2/program.pb.h"
+#include "p4_symbolic/bmv2/bmv2.pb.h"
 
 namespace p4_symbolic {
```
Comment (stefanheule) [inline p4_symbolic/bmv2/test.cc]: I don't think we want these to be in any namesapce, since this is a main executable.

**Review by stefanheule — COMMENTED** [review]
Just some minor comments.

File: p4_symbolic/bmv2/test.cc (hunk 5 above)
Comment (KinanBab) [reply p4_symbolic/bmv2/test.cc]: Yeah that makes sense

**Review by stefanheule — APPROVED** [review]

**Pull request merged**